reference/latest/src/main/java/com/example/docs/item/custom/CounterItem.java

```diff
@@ -0,0 +1,56 @@
+package com.example.docs.item.custom;
+
+import java.util.List;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.tooltip.TooltipType;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.world.World;
+
+import com.example.docs.component.ModComponents;
+
+//::1
+public class CounterItem extends Item {
+
+	public CounterItem(Settings settings) {
+		super(settings);
+	}
+	//::1
+
+	@Override
+	//::2
+	public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+		// Don't do anything on the client
+		ItemStack stack = user.getStackInHand(hand);
+		if (world.isClient) {
+			return TypedActionResult.success(stack);
+		}
+
+		// Read the current count and increase it by one
+		int count = stack.getOrDefault(ModComponents.CLICK_COUNT_COMPONENT, 0);
+		stack.set(ModComponents.CLICK_COUNT_COMPONENT, ++count);
+
+		// Return the original stack
+		return TypedActionResult.success(stack);
+	}
+	//::2
+
+	@Override
+	//::3
+	public void appendTooltip(ItemStack stack, TooltipContext context, List<Text> tooltip, TooltipType type) {
+		int count = stack.getOrDefault(ModComponents.CLICK_COUNT_COMPONENT, 0);
+		if (stack.contains(ModComponents.CLICK_COUNT_COMPONENT)) {
+			tooltip.add(Text.translatable("item.fabric-docs-reference.counter.info", count).formatted(Formatting.GOLD));
+		}
```

Comment:
Yeah I know, but I wanted to show an example of both solutions in action. I guess I can remove the `getOrDefault` and use just the if statement?

Reply:
> I guess I can remove the getOrDefault and use just the if statement?

Yes, that sounds good. Since we use `getOrDefault` in our `CounterItem#use` method below, that should make readers aware of that method (and partially imply when to use one over the other)

Reply:
Yep good idea